### DISCUSSION TITLE
New configuration values for limiting CRUD results, choosing number of results and default pagelenght

### DISCUSSION THREAD

**Discussion by mariovillani** [Laravel-Backpack/community-forum#939] [Ideas]
Showing all the results in CRUD controllers may be a slow operation (client-side, due to the rendering of datatables and server-side for the query itself).
Actually, it is possible to set the default number of rows:

`$this->crud->setDefaultPageLength(10); // number of rows shown in table view by default (client-side)`

`$this->crud->setPageLengthMenu([100, 200, 300]); // values to put in the number of results selector (client-side)`

Then, it is requested to use limit to prevent querying for too much data server side:

`$this->crud->limit(50); //limit number of results (server-side)` 

Those instructions need to be added in each CRUD controller. I think would be better to add some global configuration vars for CRUD package, setting:

- pageLenght entries
- default page length
- limit number of results

for ALL crud entities.

**Comment by karandatwani92** [comment]
Hey @MarioVillani 

Yes, you can globally set the config here `config/backpack/crud.php`
```
'defaultPageLength' => 10,
'pageLengthMenu' => [[10, 25, 50, 100, -1], [10, 25, 50, 100, 'backpack::crud.all']],
```

**Comment by mariovillani** [comment]
@karandatwani92 Correct file is `config/backpack/operations/list.php`.

I've changed it to:

```
    // How many items should be shown by default by the Datatable?
    // This value can be overwritten on a specific CRUD by calling
    // $this->crud->setDefaultPageLength(50);
    'defaultPageLength' => 10,

    // A 1D array of options which will be used for both the displayed option and the value, or
    // A 2D array in which the first array is used to define the value options and the second array the displayed options
    // If a 2D array is used, strings in the right hand array will be automatically run through trans()
    'pageLengthMenu' => [[10, 25, 50, 100], [10, 25, 50, 100]],
```

to fit my needs. As i've said, there is no server-side check made. I may alter the HTML of the select and choose -1 to show all results. A server side check **must** be done. @pxpm can you check too?

**Reply by pxpm** [reply]
@MarioVillani I was already tracking this issue and was expecting to reply to you shortly, I am going to be busy the next hour or so https://github.com/Laravel-Backpack/CRUD/pull/5503

I will finish it and merge it today. You are totally right about the server side and that can have a performance impact for all the users. 

Thanks for the heads up 🙏

**Reply by pxpm** [reply]
@MarioVillani just released https://github.com/Laravel-Backpack/CRUD/releases/tag/6.7.9 where you shouldn't be able to use any value outside the pageLengthMenu you defined. 

Ler me know if that solves your issue.

Cheers

**Reply by mariovillani** [reply]
@pxpm perfect! when trying to input a page length different from the fixed one in the config files, server is returning error 400 "Unknown page length". Thank you!
